rust/connlib/tunnel/src/tests.rs

```diff
@@ -130,6 +130,15 @@ struct ReferenceState {
     expected_dns_handshakes: VecDeque<QueryId>,
 }
 
+/// When handling DNS queries, connlib caches A as well as AAAA records, regardless of the original query.
+///
+/// To correctly model connlib's behaviour, we need to remember, whether an IP has only been cached internally or actually been returned to the client as part of a query.
+#[derive(Debug, Clone, Copy, PartialOrd, Ord, PartialEq, Eq)]
+enum IpStatus {
+    Internal,
+    Queried,
```

Comment:
```suggestion
    /// We have cached this IP but the user has not observed it
    Internal,
    /// The user has observed this IP
    Queried,
```
I had to think about it a little when reading this